### ISSUE TITLE
is possible to set delay going to next step?

### ISSUE THREAD

**Issue by juztinlazaro** [pulsardev/vue-tour#37]
Hi is possible to set a delay going to the next step?
currently, i have a scenario that the elment is not yet in DOM

**Comment by gavJackson** [comment]
+1

I also need this feature.

**Comment by Outpox** [comment]
Hi guys, have you given a look to the `startTimeout` property ?  
[See the doc for an example](https://pulsar.gitbooks.io/vue-tour/tour-configuration.html)

**Comment by goldengecko** [comment]
Thanks for the pointer - I have made it a bit more generic so that it automatically pushes a route when you need to, and then waits for it to load. 

```
    <v-tour name="introTour"
      :steps="introSteps">
      <template slot-scope="tour">
        <transition name="fade">
          <v-step v-if="tour.currentStep === index"
            v-for="(step, index) of tour.steps"
            :key="index"
            :step="step"
            :previous-step="tour.previousStep"
            :next-step="tour.nextStep"
            :stop="tour.stop"
            :is-first="tour.isFirst"
            :is-last="tour.isLast"
            :labels="tour.labels">
            <div slot="actions">
              <div class="v-step__buttons">
                <button @click.prevent="tour.stop"
                  v-if="!tour.isLast">{{ labels.buttonSkip }}</button>
                <button @click.prevent="introGoPrev(tour)"
                  v-if="!tour.isFirst">{{ labels.buttonPrevious }}</button>
                <button @click.prevent="introGoNext(tour)"
                  v-if="!tour.isLast">{{ labels.buttonNext }}</button>
                <button @click.prevent="tour.stop"
                  v-if="tour.isLast">{{ labels.buttonStop }}</button>
              </div>
            </div>
          </v-step>
        </transition>
      </template>
    </v-tour>
```
And then in the data I specify the steps where it needs to go to a specific route:
```
      introSteps: [
        {
          target: '#intro-step-0',
          header: { title: this.$t('This is a header') },
          content: this.$t('Test of tour'),
        },
        {
          route: '/services',
          target: '#intro-step-1',
          content: this.$t('Here we are'),
        },
      ],
```
And pick it up in my methods:
```
    introGoNext(tour) {
      if (tour.steps[tour.currentStep + 1].route) {
        this.$router.push(tour.steps[tour.currentStep + 1].route)
        this.$nextTick(() => {
          tour.nextStep()
        })
      } else {
        tour.nextStep()
      }
    },
    introGoPrev(tour) {
      if (tour.steps[tour.currentStep - 1].route) {
        this.$router.push(tour.steps[tour.currentStep - 1].route)
        this.$nextTick(() => {
          tour.nextStep()
        })
      } else {
        tour.nextStep()
      }
    },
```

Hope this is useful

**Comment by glorat** [comment]
It seems @pbascunana made a PR at https://github.com/pulsardev/vue-tour/pull/122

Just to comment that I solved the a problem in a different way... each of my steps has an event hook (e.g. DOM element clicked/appears) which auto-moves to that next step. Aside from solving the display issue, it also means you move to the next step automatically if the user triggered the page change themselves.

**Comment by mmorainville** [comment]
Hi all,

Actually @buwilliams has made a PR to add a `before` hook that handles asynchronous actions using Promises.
This is way more flexible than adding a param in the step with a fixed delay, that's why we've decided to merge his PR (https://github.com/pulsardev/vue-tour/pull/84) rather than the one from @pbascunana (but thanks anyway for the contribution!).

**Comment by mmorainville** [comment]
The PR has been merged and released in v1.5.0.
